qutip/parallel.py

```diff
@@ -46,6 +46,15 @@
 from qutip.ui.progressbar import BaseProgressBar, TextProgressBar
 
 
+if sys.platform == 'darwin':
+    # from 3.8 mp on mac use 'spawn', suposely better than fork, but freeze
+    # in our tests.
+    try:
+        set_start_method('fork')
```

Comment:
I believe that `multiprocessing.set_start_method` [can only be called once](https://docs.python.org/3/library/multiprocessing.html#multiprocessing.set_start_method) during any program's execution.  If we do this in `qutip`, then a user cannot call it themselves either before or after import.

If you remove the `except` catch all and try
```python
import multiprocessing
multiprocessing.set_start_method('spawn')
import qutip
```
or don't remove it and try to set the start method after the `qutip` import, you'll see the `RuntimeError`.

Instead we can do something like
```python
import multiprocessing
if sys.platform == 'darwin':
    Pool = multiprocessing.get_context('fork').Pool
else:
    Pool = multiprocessing.Pool
```
which limits the effects, and also guarantees that no matter what the caller set the start method to, we'll be using the one that works for us.

Reply:
Well seen, I did not know of `get_context`. `set_start_method` did work, but felt a little bit strong-handed, changed it.

Reply:
Multiprocessing is such a nightmare at the best of times...  I didn't know about it until recently, and I've kept my local python on 3.7 because I couldn't figure out why some of my personal multiprocessing code was no longer working on macOS python3.8.  Our lives would all be so much easier if pickling could support local function definitions or lambdas.